plasmapy/physics/distribution.py

```diff
@@ -16,17 +17,114 @@
            "Maxwellian_speed_2D",
            "Maxwellian_speed_3D",
            "kappa_velocity_1D",
-           "kappa_velocity_3D"]
+           "kappa_velocity_3D",
+           "Fermi_velocity_1D"]
+
 
 def _v_drift_units(v_drift):
     # Helper method to assign units to  v_drift if it takes a default value
     if (v_drift == 0 and
-        not isinstance(v_drift, astropy.units.quantity.Quantity)):
+            not isinstance(v_drift, astropy.units.quantity.Quantity)):
         v_drift = v_drift * u.m / u.s
     else:
         v_drift = v_drift.to(u.m / u.s)
     return v_drift
 
+
+def Fermi_velocity_1D(m,
+                      T,
+                      particle="e",
+                      vTh=np.nan,
+                      units="units",
+                      E_F=0):
+    r"""
+    Returns the Fermi-Dirac velocity distribution of a particle of mass m,
+    at temperature T with thermal velocity vTh and Fermi Energy E_F
+
+    Parameters
+    ----------
+
+    m: ~astropy.units.Quantity
+        The mass of the particle convertible to kg.
+
+    T: ~astropy.units.Quantity
+        The temperature in Kelvin.
+
+    particle: str, optional
+        Representation of the particle species(e.g., ``'p'`` for protons,
+        ``'D+'`` for deuterium, or ``'He-4 +1'`` for :math:`He_4^{+1}`
+        (singly ionized helium-4)), which defaults to electrons.
+
+    vTh: ~astropy.units.Quantity, optional
+        Thermal velocity (most probable velocity) in m/s. This is used for
+        optimization purposes to avoid re-calculating vTh, for example
+        when integrating over velocity-space.
+
+    E_F: ~astropy.units.Quantity
+        The Fermi energy of the particle in units convertible to J.
+
+    units: str, optional
+        Selects whether to run function with units and unit checks (when
+        equal to "units") or to run as unitless (when equal to "unitless").
+        The unitless version is substantially faster for intensive
+        computations.
+
+    Returns
+    -------
+    f : ~astropy.units.Quantity
+        Probability density in units of Velocity^-1, normalized so that
+        :math:`\int_{-\infty}^{+\infty} f(v) dv = 1`.
+
+    Raises
+    ------
+    TypeError
+        The parameter arguments are not Quantities and
+        cannot be converted into Quantities.
+
+    ~astropy.units.UnitConversionError
+        If the parameters are not in appropriate units.
+
+    ValueError
+        If the temperature is negative, or the particle mass or charge state
+        cannot be found.
+
+    Notes
+    -----
+
+    Examples
+    --------
+
```

Comment:
The docstring could really use a bit more of a description, a reference to the source for the formula, and some example docstrings! IPython's `%doctest_mode` is helpful for the latter`.

Reply:
Thank you, I will do improve the docstring and write unit tests. I will also implement a 2D and 3D version of it during the week